### PR TITLE
Feature: New Dashboard Page rendering a new User Profile Card and Users Completed Quests

### LINKED ISSUE #575: Dashboard V1
Loom : https://www.loom.com/share/c93d2d9c9299465cbbe245342c38884e

Figma : https://www.figma.com/file/fh0OAvj4AS08kHoSxu3DkE/%F0%9F%9A%80-Starknet-Quest?type=design&node-id=3734%3A26280&mode=design&t=EIDLigeUsWmUA8Mt-1

expected time: 3 business days

### PR DESCRIPTION
**Pull request by miiikovic**
# Pull Request type

- Feature

Resolves: #575 

## Other information

Requires data-fetch modifications. Still not ready for review

### CODE REVIEW

File: app/dashboard/page.tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+          setInitProfile(true);
+          if (hexToDecimal(addressOrDomain) === hexToDecimal(address))
+            setIsOwner(true);
+        });
+    } else {
+      setNotFound(true);
+    }
+  }, [addressOrDomain, address]);
+
+  const getIdentityData = async (id: string) => {
```
Comment (ayushtom) [inline app/dashboard/page.tsx]: can you add this in a seperate file called starknedId.service inside the `services` folder. we couple all api services in one file

File: components/UI/profileCard/profileCard.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  sinceDate,
+  userPercentile,
+  data
+}) => {
+  const [copied, setCopied] = useState(false);
+  const { data: profileData } = useStarkProfile({ address: identity?.addr });
+  const [ displayData, setDisplayData] = useState<FormattedRankingProps>([]);
+
+
+  const copyToClipboard = () => {
```
Comment (ayushtom) [inline components/UI/profileCard/profileCard.tsx:155]: i think this can be added as a util. Can be used globally if we ever need the same functionality

**Review by ayushtom — COMMENTED** [review]
You also have build issues so i couldn't review the functionality just yet

File: app/dashboard/page.tsx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+import ProfileCard from "@components/UI/profileCard/profileCard";
+import { LeaderboardTopperParams, fetchLeaderboardToppers, getBoostById, getBoosts, getCompletedQuests, getQuestById } from "@services/apiService";
+import { calculatePercentile } from "@utils/numberService";
+import { getDomainFromAddress } from "@utils/domainService";
+import BoostCard from "@components/quest-boost/boostCard";
+import { QuestDocument } from "../../types/backTypes";
+
+
+type DashboardProps = {
+  params: {
```
Comment (ayushtom) [inline app/dashboard/page.tsx:27]: i think this is not a valid way to declare page props. You might have to check the vercel docs to get the correct structure here.

**Comment by miiikovic** [comment]
User identity is undefined on the mainnet in production, which causes the application to render "Loading profile data...". It works fine on localhost. Sepolia is good in both cases.

**Comment by ayushtom** [comment]
continuing on a new branch to improve the PR

**Comment by ayushtom** [comment]
new branch here https://github.com/starknet-id/starknet.quest/pull/599

**Pull request merged**